src/meta/src/stream/stream_manager.rs

```diff
@@ -705,6 +710,88 @@ impl GlobalStreamManager {
 
         Ok(())
     }
+
+    async fn trigger_scale_out(&self, workers: Vec<WorkerId>) -> MetaResult<()> {
+        let _reschedule_job_lock = self.reschedule_lock.write().await;
+
+        let fragment_worker_changes = {
+            let guard = self.fragment_manager.get_fragment_read_guard().await;
+            let mut fragment_worker_changes = HashMap::new();
+            for table_fragments in guard.table_fragments().values() {
+                for fragment_id in table_fragments.fragment_ids() {
+                    fragment_worker_changes.insert(
+                        fragment_id,
+                        PbWorkerChanges {
+                            include_worker_ids: workers.clone(),
+                            ..Default::default()
+                        },
+                    );
+                }
+            }
+            fragment_worker_changes
+        };
+
+        let reschedules = self
+            .scale_controller
+            .generate_stable_resize_plan(
+                StableResizePolicy {
+                    fragment_worker_changes,
+                },
+                None,
+            )
+            .await?;
+
+        self.reschedule_actors(
+            reschedules,
+            RescheduleOptions {
+                resolve_no_shuffle_upstream: true,
+            },
+        )
+        .await?;
+
+        Ok(())
+    }
+
+    async fn run(&self, mut shutdown_rx: Receiver<()>) {
+        let (local_notification_tx, mut local_notification_rx) =
+            tokio::sync::mpsc::unbounded_channel();
+
+        self.env
+            .notification_manager()
+            .insert_local_sender(local_notification_tx)
+            .await;
+
+        loop {
+            tokio::select! {
+                biased;
+
+                _ = &mut shutdown_rx => {
+                    tracing::info!("Barrier manager is stopped");
+                    break;
+                }
+
+                notification = local_notification_rx.recv() => {
+                    let notification = notification.expect("local notification channel closed in loop of stream manager");
+                    if let LocalNotification::WorkerNodeActivated(worker) = &notification {
```

Comment:
Actually, whether it is a new node or a node already in the cluster, when they connect to meta and activate themselves, this notification will be triggered. How about adding a filtering condition, if there is already an actor running on this CN, it will not trigger scale out.

Reply:
When you receive the activate notification, the worker is already exist in the cluster manager. I think you need to read from the fragment manager to check whether there are any actors that are placed in the worker. Just like how we identify the new nodes in migration.